internal/pkg/deploy/cloudformation/cloudformation.go

```diff
@@ -172,14 +172,38 @@ func (cf CloudFormation) errorEvents(stackName string) ([]string, error) {
 	return reasons, nil
 }
 
-type renderStackChangesInput struct {
+type executeAndRenderChangeSetInput struct {
 	stackName        string
 	stackDescription string
 	createChangeSet  func() (string, error)
 }
 
-func (cf CloudFormation) newRenderWorkloadInput(w progress.FileWriter, stack *cloudformation.Stack) *renderStackChangesInput {
-	in := &renderStackChangesInput{
+func (cf CloudFormation) newCreateChangeSetInput(w progress.FileWriter, stack *cloudformation.Stack) *executeAndRenderChangeSetInput {
+	in := &executeAndRenderChangeSetInput{
+		stackName:        stack.Name,
+		stackDescription: fmt.Sprintf("Creating the infrastructure for stack %s", stack.Name),
+	}
+	in.createChangeSet = func() (changeSetID string, err error) {
+		spinner := progress.NewSpinner(w)
+		label := fmt.Sprintf("Proposing infrastructure changes for stack %s", stack.Name)
+		spinner.Start(label)
+
+		var errAlreadyExists *cloudformation.ErrStackAlreadyExists
+		changeSetID, err = cf.cfnClient.Create(stack)
+		switch {
+		case err != nil && !errors.As(err, &errAlreadyExists):
+			spinner.Stop(log.Serrorf("%s\n", label))
+			return "", cf.handleStackError(stack.Name, err)
+		default:
+			spinner.Stop(log.Ssuccessf("%s\n", label))
+			return changeSetID, err
+		}
```

Comment:
nit (i <img src="https://slackmojis.com/emojis/10804-meow_love/image/1643515051/meow_love.png" data-canonical-src="https://slackmojis.com/emojis/10804-meow_love/image/1643515051/meow_love.png" width="15" height="15" /> `switch` but there's only one condition 🙄)
```suggestion
		if err != nil && !errors.As(err, &errAlreadyExists) {
		    spinner.Stop(log.Serrorf("%s\n", label))
			return "", cf.handleStackError(stack.Name, err)
		}
		
		spinner.Stop(log.Ssuccessf("%s\n", label))
		return changeSetID, err
```

Reply:
<img src="https://emojis.slackmojis.com/emojis/images/1643514288/2583/homer_doh.png?1643514288" width="30px"/> derp, thanks! fixed